### DISCUSSION TITLE
using route with parameters breaks logic of model injection in Screen

### DISCUSSION THREAD

**Discussion by banchap** [orchidsoftware/platform#2796] [Q&A]
hello I'm trying to use route with parameters in platform.php like
`Route::screen('project/{projectId}/episode', MyScreen::class)`
but when I update data using async method 
```
return ModalToggle::make('Изменить')
    ->modal('episodeModalEdit')
    ->method('update')
    ->icon('note')
    ->asyncParameters([
        'episode' => $episode->id
    ]);
```
I change element with ID of {projectId} from route not episode id from asyncParameters
update method gets $episode with wrong id:
```
public function update(Episode $episode, Request $request): void
{
    // update stuff
}
```

i checked code and found that in src/Screen/Resolvers/ScreenDependencyResolver.php id of model is getting by first argument

`$value = $httpQueryArguments->shift();`

so if I'm not mistaken the key (id) of dependent model is getting as any first argument from the route params or the request params

is it correct behavior or it should be something like 
 `$value = $httpQueryArguments->get($parameter->getName()) ?? $httpQueryArguments->shift();`
right now i don't see any reason using associative arrays in asyncParameters() parameter

**Comment by tabuna** [comment]
Hello, it seems like you are using an outdated version of the package, as the file `/Screen/Resolvers/ScreenDependencyResolver.php` is missing. Please consider updating it to resolve this issue.

**Reply by banchap** [reply]
Hi! Thank you for respond. I've tried to update but still have the same problem but in quite a different way.
The problem is that it takes parameter projectId which is not related directly to the model i'm changing. I found workaround by setting action directly to Modal like
```
return ModalToggle::make('Edit')
    ->modal('episodeModalEdit')
    ->action('/admin/episodes/createOrUpdate') // removed project id in url here
    ->method('update')
    ->icon('note')
    ->asyncParameters([
        'episode' => $episode->id
    ]);
```
but it seems little weird because i need to setup additional root admin/episodes to get it work this way
Maybe I'm missing something, but it seems to me that Orchid should not get any first parameter from route as model id blindly
